CalibTracker/SiPhase2TrackerESProducers/plugins/SiPhase2BadStripConfigurableFakeESSource.cc

```diff
@@ -0,0 +1,255 @@
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+
+// neede for the random number generation
+#include "CLHEP/Random/RandFlat.h"
+#include "CLHEP/Random/JamesRandom.h"
+
+class SiPhase2BadStripConfigurableFakeESSource : public edm::ESProducer, public edm::EventSetupRecordIntervalFinder {
+public:
+  SiPhase2BadStripConfigurableFakeESSource(const edm::ParameterSet&);
+  ~SiPhase2BadStripConfigurableFakeESSource() override = default;
+
+  void setIntervalFor(const edm::eventsetup::EventSetupRecordKey&,
+                      const edm::IOVSyncValue& iov,
+                      edm::ValidityInterval& iValidity) override;
+
+  typedef std::unique_ptr<SiStripBadStrip> ReturnType;
+  ReturnType produce(const SiPhase2OuterTrackerBadStripRcd&);
+
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+private:
+  std::map<unsigned short, unsigned short> clusterizeBadChannels(
+      const std::vector<Phase2TrackerDigi::PackedDigiType>& maskedChannels);
+
+  // configurables
+  bool m_printDebug;
+  float m_badComponentsFraction;
+
+  // random engine
+  std::unique_ptr<CLHEP::HepRandomEngine> m_engine;
+
+  // es tokens
+  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> m_trackTopoToken;
+  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> m_geomToken;
+};
+
+SiPhase2BadStripConfigurableFakeESSource::SiPhase2BadStripConfigurableFakeESSource(const edm::ParameterSet& iConfig)
+    : m_engine(new CLHEP::HepJamesRandom(iConfig.getParameter<unsigned int>("seed"))) {
+  auto cc = setWhatProduced(this);
+  m_trackTopoToken = cc.consumes();
+  m_geomToken = cc.consumes();
+
+  m_printDebug = iConfig.getUntrackedParameter<bool>("printDebug", false);
+  m_badComponentsFraction = iConfig.getParameter<double>("badComponentsFraction");
+
+  if (m_badComponentsFraction > 1. || m_badComponentsFraction < 0.) {
+    throw cms::Exception("Inconsistent configuration")
+        << "[SiPhase2BadStripChannelBuilder::c'tor] the requested fraction of bad components is unphysical. \n";
+  }
+
+  findingRecord<SiPhase2OuterTrackerBadStripRcd>();
+}
+
+void SiPhase2BadStripConfigurableFakeESSource::setIntervalFor(const edm::eventsetup::EventSetupRecordKey&,
+                                                              const edm::IOVSyncValue& iov,
+                                                              edm::ValidityInterval& iValidity) {
+  iValidity = edm::ValidityInterval{iov.beginOfTime(), iov.endOfTime()};
+}
+
+// ------------ method called to produce the data  ------------
+SiPhase2BadStripConfigurableFakeESSource::ReturnType SiPhase2BadStripConfigurableFakeESSource::produce(
+    const SiPhase2OuterTrackerBadStripRcd& iRecord) {
+  using namespace edm::es;
+  using Phase2TrackerGeomDetUnit = PixelGeomDetUnit;
+
+  //TrackerTopology const& tTopo = iRecord.get(m_trackTopoToken);
```

Comment:
Is this commented out code needed?

Reply:
prefer to keep, as the dependent record explicitly allows for such dependency (cf.: https://github.com/cms-sw/cmssw/blob/0905a2167a02f4d820a67d5bf1b619d5e75af280/CondFormats/DataRecord/interface/SiPhase2OuterTrackerCondDataRecords.h#L19). 
At the moment is not used as the `ESSource` filling logic is quite simple and randomly distributed bad components flatly in detector space, though at some point we might want to move to something that necessitates the knowledge of the Topology tree.